doc/source/user_guide/sparse.rst

```diff
@@ -105,84 +130,117 @@ keeps an arrays of all of the locations where the data are not equal to the
 fill value. The ``block`` format tracks only the locations and sizes of blocks
 of data.
 
-.. _sparse.dtype:
+.. _sparse.calculation:
+
+Sparse Calculation
+------------------
 
-Sparse Dtypes
--------------
+You can apply NumPy *ufuncs* to ``SparseArray`` and get a ``SparseArray`` as a result.
 
-Sparse data should have the same dtype as its dense representation. Currently,
-``float64``, ``int64`` and ``bool`` dtypes are supported. Depending on the original
-dtype, ``fill_value`` default changes:
+.. ipython:: python
+
+   arr = pd.SparseArray([1., np.nan, np.nan, -2., np.nan])
+   np.abs(arr)
 
-* ``float64``: ``np.nan``
-* ``int64``: ``0``
-* ``bool``: ``False``
+
+The *ufunc* is also applied to ``fill_value``. This is needed to get
+the correct dense result.
 
 .. ipython:: python
 
-   s = pd.Series([1, np.nan, np.nan])
-   s
-   s.to_sparse()
+   arr = pd.SparseArray([1., -1, -1, -2., -1], fill_value=-1)
+   np.abs(arr)
+   np.abs(arr).to_dense()
 
-   s = pd.Series([1, 0, 0])
-   s
-   s.to_sparse()
+.. _sparse.migration:
 
-   s = pd.Series([True, False, True])
-   s
-   s.to_sparse()
+Migrating
+---------
+
+In older versions of pandas, the ``SparseSeries`` and ``SparseDataFrame`` classes (documented below)
+were the preferred way to work with sparse data. With the advent of extension arrays, these subclasses
+are no longer needed. Their purpose is better served by using a regular Series or DataFrame with
+sparse values instead.
+
+**There's no performance or memory penalty to using a Series or DataFrame with sparse values,
+rather than a SparseSeries or SparseDataFrame**.
+
+This section provides some guidance on migrating your code to the new style. As a reminder, you can
+use the python warnings module to control warnings. If you wish to ignore the warnings,
+
+.. code-block:: python
+
+   >>> import warnings
+
+   >>> warnings.filterwarnings('ignore', 'Sparse', FutureWarning)
+   >>> pd.SparseSeries()  # No warning message
+   Series([], dtype: Sparse[float64, nan])
+   BlockIndex
+   Block locations: array([], dtype=int32)
+   Block lengths: array([], dtype=int32)
+
+But we recommend modifying your code, rather than ignoring the warning.
+
+**Construction**
 
-You can change the dtype using ``.astype()``, the result is also sparse. Note that
-``.astype()`` also affects to the ``fill_value`` to keep its dense representation.
+From an array-like, use the regular :class:`Series` or
+:class:`DataFrame` constructors with :class:`SparseArray` values.
 
+.. code-block:: python
+
+   # Old way
+   >>> pd.SparseDataFrame({"A": [0, 1]})
 
 .. ipython:: python
 
-   s = pd.Series([1, 0, 0, 0, 0])
-   s
-   ss = s.to_sparse()
-   ss
-   ss.astype(np.float64)
+   # New way
+   pd.DataFrame({"A": pd.SparseArray([0, 1])})
 
-It raises if any value cannot be coerced to specified dtype.
+From a SciPy sparse matrix, use :meth:`DataFrame.sparse.from_spmatrix`,
```

Comment:
ref to scipy docs?

Reply:
I was thinking, it might also be good to somewhere in this document note the difference between both: when creating from sparse matrix, you convert the sparse values in several 1D sparse columns, not a single 2D sparse matrix

Reply:
Can you clarify this @jorisvandenbossche

Reply:
People might think that, since you convert the sparse matrix in a pandas DataFrame, that both are similar and a sparse dataframe is kind of a wrapper around sparse matrix with column names and indices. But the way it is stored is quite different: in a sparse DataFrame each column is stored separately as a sparse 1D array (although compared to COO it is not that much different)

Reply:
Gotcha. I'll note that the data will be copied, and the similar structure to COO.

Reply:
Really, we need a new "Interaction with SciPy.sparse" section, that is dataframe-focused.